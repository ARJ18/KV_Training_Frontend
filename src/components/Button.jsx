/* eslint-disable no-unused-vars */
/* eslint-disable react/prop-types */
import "../styles/Button.css";
import React from "react";
const Button = ({ label, handleClick, color }) => {
  return (
    <button
      onClick={handleClick}
      className={color}
      data-testid="button-test-id"
    >
      {label}
    </button>
  );
};

export default Button;
