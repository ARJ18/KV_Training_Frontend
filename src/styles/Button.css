.blue {
  padding: 1rem;
  border: none;
  color: white;
  width: auto;
  cursor: pointer;
  max-height: 50px;
  font-size: 16px;
  background: #03aeee 0% 0% no-repeat padding-box;
  box-shadow: 0px 3px 6px #bfeeff;
  border-radius: 3px;
  opacity: 1;
  margin: 8px;
}

.white {
  padding: 1rem;
  border: none;
  color: rgb(0, 0, 0);
  width: auto;
  cursor: pointer;
  margin: 8px;
  max-height: 50px;
  font-size: 16px;
  background: #ffffff 0% 0% no-repeat padding-box;
  box-shadow: 0px 3px 6px #c9c9c929;
  border: 1px solid #8a8c8f;
  border-radius: 3px;
  opacity: 1;
}
.login-button:hover {
  background-color: #067bbe;
}
