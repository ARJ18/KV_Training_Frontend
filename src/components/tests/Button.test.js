/* eslint-disable no-undef */
import Button from "../Button";
import { render ,fireEvent} from "@testing-library/react";

import React from "react";

describe("Button test cases", () => {
  test("If button is rendered successfully", () => {
    const { getByTestId } = render(<Button />);
    expect(getByTestId('button-test-id')).toBeTruthy();

  })
  test('If handleClick is triggered correctly', () => { 
    const handleClick = jest.fn();
    const {getByTestId} = render(<Button handleClick={handleClick}/>);
    fireEvent.click(getByTestId("button-test-id"));
    expect(handleClick).not.toHaveBeenCalledTimes(2);
   })
   test('If snapshot is matching',()=>{
    const handleClick = jest.fn();
    const {baseElement} = render(<Button label="button" handleClick={handleClick} color="blue"/>);
    expect(baseElement).toMatchSnapshot();
   })
});

