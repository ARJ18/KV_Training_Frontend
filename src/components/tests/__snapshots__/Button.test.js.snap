// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test cases If snapshot is matching 1`] = `
<body>
  <div>
    <button
      class="blue"
      data-testid="button-test-id"
    >
      button
    </button>
  </div>
</body>
`;
